Compiler/src/profiling.jl

```diff
@@ -0,0 +1,38 @@
+const WITH_ITTAPI = ccall(:jl_ittapi_enabled, Cint, ()) != 0
+const WITH_TRACY = ccall(:jl_tracy_enabled, Cint, ()) != 0
+
+# Maybe create a singleton type TimingLock so one can use `@lock`
+lock_timing() = ccall(:jl_timing_lock_lock, Cvoid, ())
+unlock_timing() = ccall(:jl_timing_lock_unlock, Cvoid, ())
+
+include("profiling/tracy.jl")
+include("profiling/ittapi.jl")
+
+if WITH_TRACY || WITH_ITTAPI
+    macro zone(name, ex::Expr)
+        lock_timing()
```

Comment:
This lock should probably be taken just in the implementation of `tracy_zone_create`